### ISSUE TITLE
PojoComponentTuplizer error with Spring Data JPA

### ISSUE THREAD

**Issue by MarcelCoding** [spring-projects-experimental/spring-native#603]
If I boot with spring data jpa, i will get the following error: (https://github.com/MarcelCoding/luna/tree/spring-native - entities are in `luna-cacti`)
```bash
sudo docker run --rm -e SQL_HOST=192.168.178.28 -e SQL_USER=root -e SQL_PASS=root -p 8080:8080 luna-main:edge 
```
```
2021-03-12 15:44:38.716  INFO 1 --- [           main] o.s.nativex.NativeListener               : This application is bootstrapped with code generated with Spring AOT

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::       (v2.4.4-SNAPSHOT)

2021-03-12 15:44:38.717  INFO 1 --- [           main] com.github.marcelcoding.luna.Bootstrap   : Starting Bootstrap using Java 11.0.10 on 6902d60124d1 with PID 1 (/workspace/com.github.marcelcoding.luna.Bootstrap started by cnb in /workspace)
2021-03-12 15:44:38.717  INFO 1 --- [           main] com.github.marcelcoding.luna.Bootstrap   : No active profile set, falling back to default profiles: default
2021-03-12 15:44:38.733  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2021-03-12 15:44:38.736  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 2 ms. Found 6 JPA repository interfaces.
2021-03-12 15:44:38.758  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2021-03-12 15:44:38.764  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2021-03-12 15:44:38.797 ERROR 1 --- [           main] j.LocalContainerEntityManagerFactoryBean : Failed to initialize JPA EntityManagerFactory: [PersistenceUnit: default] Unable to build Hibernate SessionFactory; nested exception is org.hibernate.HibernateException: Unable to instantiate default tuplizer [org.hibernate.tuple.component.PojoComponentTuplizer]
2021-03-12 15:44:38.797  WARN 1 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory; nested exception is org.hibernate.HibernateException: Unable to instantiate default tuplizer [org.hibernate.tuple.component.PojoComponentTuplizer]
2021-03-12 15:44:38.797  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2021-03-12 15:44:38.797  WARN 1 --- [           main] o.s.b.f.support.DisposableBeanAdapter    : Destroy method 'close' on bean with name 'dataSource' threw an exception: java.lang.Error: java.lang.IllegalAccessException: Cannot set final field: java.util.concurrent.CopyOnWriteArrayList.lock. Enable by specifying "allowWrite" for this field in the reflection configuration.
2021-03-12 15:44:38.798  INFO 1 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-03-12 15:44:38.798 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory; nested exception is org.hibernate.HibernateException: Unable to instantiate default tuplizer [org.hibernate.tuple.component.PojoComponentTuplizer]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1786) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[na:na]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[na:na]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:769) ~[com.github.marcelcoding.luna.Bootstrap:2.4.4-SNAPSHOT]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:761) ~[com.github.marcelcoding.luna.Bootstrap:2.4.4-SNAPSHOT]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426) ~[com.github.marcelcoding.luna.Bootstrap:2.4.4-SNAPSHOT]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:326) ~[com.github.marcelcoding.luna.Bootstrap:2.4.4-SNAPSHOT]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1313) ~[com.github.marcelcoding.luna.Bootstrap:2.4.4-SNAPSHOT]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302) ~[com.github.marcelcoding.luna.Bootstrap:2.4.4-SNAPSHOT]
        at com.github.marcelcoding.luna.Bootstrap.main(Bootstrap.java:36) ~[com.github.marcelcoding.luna.Bootstrap:na]
Caused by: javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory; nested exception is org.hibernate.HibernateException: Unable to instantiate default tuplizer [org.hibernate.tuple.component.PojoComponentTuplizer]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:421) ~[com.github.marcelcoding.luna.Bootstrap:5.3.5-SNAPSHOT]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:396) ~[com.github.marcelcoding.luna.Bootstrap:5.3.5-SNAPSHOT]
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:341) ~[com.github.marcelcoding.luna.Bootstrap:5.3.5-SNAPSHOT]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[na:na]
        ... 16 common frames omitted
Caused by: org.hibernate.HibernateException: Unable to instantiate default tuplizer [org.hibernate.tuple.component.PojoComponentTuplizer]
        at org.hibernate.tuple.component.ComponentTuplizerFactory.constructTuplizer(ComponentTuplizerFactory.java:107) ~[na:na]
        at org.hibernate.tuple.component.ComponentTuplizerFactory.constructDefaultTuplizer(ComponentTuplizerFactory.java:128) ~[na:na]
        at org.hibernate.tuple.component.ComponentMetamodel.<init>(ComponentMetamodel.java:77) ~[na:na]
        at org.hibernate.tuple.component.ComponentMetamodel.<init>(ComponentMetamodel.java:52) ~[na:na]
        at org.hibernate.mapping.Component.getType(Component.java:227) ~[na:na]
        at org.hibernate.mapping.Property.getType(Property.java:70) ~[na:na]
        at org.hibernate.jpa.event.internal.CallbackBuilderLegacyImpl.resolveEmbeddableCallbacks(CallbackBuilderLegacyImpl.java:227) ~[na:na]
        at org.hibernate.jpa.event.internal.CallbackBuilderLegacyImpl.buildCallbacksForEmbeddable(CallbackBuilderLegacyImpl.java:79) ~[na:na]
        at org.hibernate.event.service.internal.EventListenerRegistryImpl.prepare(EventListenerRegistryImpl.java:154) ~[na:na]
        at org.hibernate.boot.internal.MetadataImpl.initSessionFactory(MetadataImpl.java:379) ~[na:na]
        at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:214) ~[na:na]
        at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:468) ~[na:na]
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1259) ~[na:na]
        at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:58) ~[na:na]
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:365) ~[com.github.marcelcoding.luna.Bootstrap:5.3.5-SNAPSHOT]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:409) ~[com.github.marcelcoding.luna.Bootstrap:5.3.5-SNAPSHOT]
        ... 20 common frames omitted
Caused by: java.lang.NullPointerException: null
        at org.hibernate.tuple.component.ComponentTuplizerFactory.constructTuplizer(ComponentTuplizerFactory.java:104) ~[na:na]
        ... 35 common frames omitted

```

**Comment by abelsromero** [comment]
I am also having issues running native+jpa, but the error is different.
Here is a simple reproducer https://github.com/abelsromero/springboot-native-jp (see README for details). It runs fine in Java, but when run in native it fails with error below. It seems it could be fixed with some extra configuration for graalvm, but I think this should be provided by spring boot right?

Note that to compile I had to add `org.hibernate.dialect.PostgreSQL95Dialect` to `META-INF/native-image`.

```
~/temp/springboot >>> ./com.example.springboot.SpringbootApplication                                                                                                                                                                                                                                                                                     ±[●][main]
22:27:15.526 [main] INFO org.hibernate.dialect.Dialect - HHH000400: Using dialect: org.hibernate.dialect.PostgreSQL95Dialect
2021-03-17 23:27:15.532  INFO 185805 --- [           main] o.s.nativex.NativeListener               : This application is bootstrapped with code generated with Spring AOT

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.4.3)

2021-03-17 23:27:15.532  INFO 185805 --- [           main] c.e.springboot.SpringbootApplication     : Starting SpringbootApplication using Java 11.0.10 on man-pc with PID 185805 (/home/asalgadr/temp/springboot/com.example.springboot.SpringbootApplication started by asalgadr in /home/asalgadr/temp/springboot)
2021-03-17 23:27:15.532  INFO 185805 --- [           main] c.e.springboot.SpringbootApplication     : No active profile set, falling back to default profiles: default
2021-03-17 23:27:15.557  INFO 185805 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2021-03-17 23:27:15.558  INFO 185805 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 0 ms. Found 1 JPA repository interfaces.
2021-03-17 23:27:15.572  INFO 185805 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
Mar 17, 2021 11:27:15 PM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-nio-8080"]
Mar 17, 2021 11:27:15 PM org.apache.catalina.core.StandardService startInternal
INFO: Starting service [Tomcat]
Mar 17, 2021 11:27:15 PM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet engine: [Apache Tomcat/9.0.43]
Mar 17, 2021 11:27:15 PM org.apache.catalina.core.ApplicationContext log
INFO: Initializing Spring embedded WebApplicationContext
2021-03-17 23:27:15.573  INFO 185805 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 40 ms
2021-03-17 23:27:15.587  INFO 185805 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: default]
2021-03-17 23:27:15.587  INFO 185805 --- [           main] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 5.4.28.Final
2021-03-17 23:27:15.587  INFO 185805 --- [           main] o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
2021-03-17 23:27:15.588  INFO 185805 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2021-03-17 23:27:15.592  INFO 185805 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2021-03-17 23:27:15.592  INFO 185805 --- [           main] org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.PostgreSQL95Dialect
2021-03-17 23:27:15.605  WARN 185805 --- [           main] o.h.id.enhanced.OptimizerFactory         : HHH000322: Unable to instantiate specified optimizer [none], falling back to noop
2021-03-17 23:27:15.638  INFO 185805 --- [           main] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
2021-03-17 23:27:15.638  INFO 185805 --- [           main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
2021-03-17 23:27:15.658  WARN 185805 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'filmController': Unsatisfied dependency expressed through field 'filmRepository'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'filmRepository' defined in com.example.springboot.films.FilmRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface com.example.springboot.films.FilmRepository, interface org.springframework.data.repository.Repository, interface org.springframework.transaction.interceptor.TransactionalProxy, interface org.springframework.aop.framework.Advised, interface org.springframework.core.DecoratingProxy] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
2021-03-17 23:27:15.658  INFO 185805 --- [           main] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2021-03-17 23:27:15.658  INFO 185805 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2021-03-17 23:27:15.659  WARN 185805 --- [           main] o.s.b.f.support.DisposableBeanAdapter    : Destroy method 'close' on bean with name 'dataSource' threw an exception: java.lang.Error: java.lang.IllegalAccessException: Cannot set final field: java.util.concurrent.CopyOnWriteArrayList.lock. Enable by specifying "allowWrite" for this field in the reflection configuration.
Mar 17, 2021 11:27:15 PM org.apache.catalina.core.StandardService stopInternal
INFO: Stopping service [Tomcat]
2021-03-17 23:27:15.659  INFO 185805 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-03-17 23:27:15.660 ERROR 185805 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'filmController': Unsatisfied dependency expressed through field 'filmRepository'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'filmRepository' defined in com.example.springboot.films.FilmRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface com.example.springboot.films.FilmRepository, interface org.springframework.data.repository.Repository, interface org.springframework.transaction.interceptor.TransactionalProxy, interface org.springframework.aop.framework.Advised, interface org.springframework.core.DecoratingProxy] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:660) ~[na:na]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640) ~[na:na]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[na:na]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[com.example.springboot.SpringbootApplication:5.3.4]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[na:na]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[na:na]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:944) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:917) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:582) ~[na:na]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144) ~[na:na]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:767) ~[com.example.springboot.SpringbootApplication:2.4.3]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759) ~[com.example.springboot.SpringbootApplication:2.4.3]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426) ~[com.example.springboot.SpringbootApplication:2.4.3]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:326) ~[com.example.springboot.SpringbootApplication:2.4.3]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1311) ~[com.example.springboot.SpringbootApplication:2.4.3]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300) ~[com.example.springboot.SpringbootApplication:2.4.3]
        at com.example.springboot.SpringbootApplication.main(SpringbootApplication.java:12) ~[com.example.springboot.SpringbootApplication:na]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'filmRepository' defined in com.example.springboot.films.FilmRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface com.example.springboot.films.FilmRepository, interface org.springframework.data.repository.Repository, interface org.springframework.transaction.interceptor.TransactionalProxy, interface org.springframework.aop.framework.Advised, interface org.springframework.core.DecoratingProxy] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1786) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[na:na]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[na:na]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[na:na]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[na:na]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[na:na]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657) ~[na:na]
        ... 21 common frames omitted
Caused by: com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface com.example.springboot.films.FilmRepository, interface org.springframework.data.repository.Repository, interface org.springframework.transaction.interceptor.TransactionalProxy, interface org.springframework.aop.framework.Advised, interface org.springframework.core.DecoratingProxy] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
        at com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:87) ~[na:na]
        at com.oracle.svm.reflect.proxy.DynamicProxySupport.getProxyClass(DynamicProxySupport.java:113) ~[na:na]
        at java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:66) ~[na:na]
        at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1006) ~[na:na]
        at org.springframework.aop.framework.JdkDynamicAopProxy.getProxy(JdkDynamicAopProxy.java:126) ~[na:na]
        at org.springframework.aop.framework.ProxyFactory.getProxy(ProxyFactory.java:110) ~[na:na]
        at org.springframework.data.repository.core.support.RepositoryFactorySupport.getRepository(RepositoryFactorySupport.java:309) ~[na:na]
        at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.lambda$afterPropertiesSet$5(RepositoryFactoryBeanSupport.java:323) ~[com.example.springboot.SpringbootApplication:2.4.5]
        at org.springframework.data.util.Lazy.getNullable(Lazy.java:230) ~[na:na]
        at org.springframework.data.util.Lazy.get(Lazy.java:114) ~[na:na]
        at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.afterPropertiesSet(RepositoryFactoryBeanSupport.java:329) ~[com.example.springboot.SpringbootApplication:2.4.5]
        at org.springframework.data.jpa.repository.support.JpaRepositoryFactoryBean.afterPropertiesSet(JpaRepositoryFactoryBean.java:144) ~[com.example.springboot.SpringbootApplication:2.4.5]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[na:na]
        ... 31 common frames omitted


```

**Comment by schauder** [comment]
@abelsromero could you create a separate issue for that please?

**Comment by abelsromero** [comment]
> @abelsromero could you create a separate issue for that please?

Done, many thanks #630

**Comment by sdeleuze** [comment]
@MarcelCoding I can't make work your `spring-native` branch, please check it is in good state to reproduce and which command we should run exactly to reproduce (since that's a multi module project).

**Comment by MarcelCoding** [comment]
@sdeleuze I removed all stuff, witch has nothing to do with the error.
to run it:
```bash
sudo ./gradlew bootBuildImage
sudo docker run --rm -e SQL_HOST=192.168.178.28 -e SQL_USER=root -e SQL_PASS=root luna-main:edge 
```

**Comment by sdeleuze** [comment]
@MarcelCoding Could you please provide a `docker-compose.yml` that allows me to run this sample without a local database ?

**Comment by MarcelCoding** [comment]
@sdeleuze I added one to the repository.

**Comment by sdeleuze** [comment]
There is a remaining `Proxy class defined by interfaces [interface com.github.marcelcoding.luna.cacti.repository.CareGroupRepository, interface org.springframework.data.repository.Repository, interface org.springframework.transaction.interceptor.TransactionalProxy, interface org.springframework.aop.framework.Advised, interface org.springframework.core.DecoratingProxy] not found.` error but let's fix that as part of #630.

**Comment by MarcelCoding** [comment]
Ok, thank you.
